src/redux/reducers/HydrantReducer.js

```diff
@@ -27,9 +27,12 @@ function updateHydrantFeatures(hydrant, editedFields) {
   if (editedFields.coords) {
     feature.getGeometry().setCoordinates(Projection.fromLonLat(editedFields.coords));
   }
-  if (_.has(editedFields, 'trail') && feature.get('selected')) {
+  if (editedFields.trail === null) {
     feature.unset('selected');
   }
+  if (_.has(editedFields, 'trail') && feature.get('selected')) {
+    // Use if we decide to allow hydrant trail reassignment.
```

Comment:
Why change this part? If a hydrant's trail was changed, whether through removal or changing it, the feature should deselect. 

The only difference I can tell is that now when a hydrant gets assigned a trail from not having one (or reassigned a new trail), it won't be unselected if it had that state. I would expect it to unselect, no?